src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialServiceSupport.scala

```diff
@@ -57,13 +62,19 @@ trait TrialServiceSupport extends LazyLogging {
     val projects: Seq[TrialProject] = trialDAO.projectReport
     logger.info(s"makeSpreadsheetValues processing ${projects.length} projects ...")
 
-    // find the Thurloe KVPs for any users referenced in those projects
-    // TODO: this should switch to using Thurloe's bulk endpoint, so we're not making a large burst of requests.
-    val profileWrappers:Future[Seq[ProfileWrapper]] = Future.sequence(projects.filter(p => p.user.isDefined).map { p =>
-      thurloeDAO.getAllKVPs(p.user.get.userSubjectId, managerInfo) map { kvpOption =>
-        kvpOption.getOrElse(ProfileWrapper(p.user.get.userSubjectId, List.empty[FireCloudKeyValue]))
+    // from the list of projects, get assigned userids
+    val assignedUsers:List[String] = projects.toList.filter(p => p.user.isDefined).flatMap(_.user.map(_.userSubjectId))
+    logger.info(s"makeSpreadsheetValues found ${assignedUsers.length} users assigned to projects ...")
+
+    // split users into chunks of size N for efficient querying to Thurloe
+    val profileQueries = assignedUsers
```

Comment:
yes, that's exactly what it will do. This is the same behavior as before this PR; only the quantity+quality of the queries has changed, but they're still shot off in one big eager batch. Also see the `TODO` a few lines below ...